### PR TITLE
Add MoneyDoorPlugin

### PR DESCRIPTION
**Pull request by MrCredulous**
Hi Lue Li,

I added a new Plugin in our code, would you want to merge it with master branch?

### CODE REVIEW

File: src/main/java/edu/cmu/cs/cs214/rec10/plugin/MoneyDoorPlugin.java [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+import edu.cmu.cs.cs214.rec10.framework.core.GamePlugin;
+import edu.cmu.cs.cs214.rec10.framework.core.Player;
+
+import java.util.*;
+
+public class MoneyDoorPlugin implements GamePlugin {
+
+    private int[] arr = {-3, -2, -1, 0, 1, 2, 3, 4, 5, 6};
+    private int ONE = 1;
+    private int TEN = 10;
```
Comment (lueliCMU) [inline src/main/java/edu/cmu/cs/cs214/rec10/plugin/MoneyDoorPlugin.java]: should be a final static

File: src/main/java/edu/cmu/cs/cs214/rec10/plugin/MoneyDoorPlugin.java [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+
+    public int getGridWidth() {
+        return TEN;
+    }
+
+    public int getGridHeight() {
+        return ONE;
+    }
+
+    public void onRegister(GameFramework fwk) {
```
Comment (lueliCMU) [inline src/main/java/edu/cmu/cs/cs214/rec10/plugin/MoneyDoorPlugin.java]: add JavaDoc for those methods

File: src/main/java/edu/cmu/cs/cs214/rec10/plugin/MoneyDoorPlugin.java [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+        this.framework = fwk;
+    }
+
+    public void onNewGame() {
+        // Make random permutation of the array
+        Random random = new Random();
+
+        int i;
```
Comment (lueliCMU) [inline src/main/java/edu/cmu/cs/cs214/rec10/plugin/MoneyDoorPlugin.java]: better to be private

File: src/main/java/edu/cmu/cs/cs214/rec10/plugin/MoneyDoorPlugin.java [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+        }
+//System.out.println("game is over");
+        return true;
+    }
+
+    public String getGameOverMessage() {
+        if (PlayerScores.get("X") > PlayerScores.get("O")) {
+            return "Congratulations X with a score of " + PlayerScores.get("X") + " !";
+        } else {
+            return "Congratulations O with a score of " + PlayerScores.get("O") + " !";
```
Comment (lueliCMU) [inline src/main/java/edu/cmu/cs/cs214/rec10/plugin/MoneyDoorPlugin.java]: s

File: src/main/java/edu/cmu/cs/cs214/rec10/plugin/MoneyDoorPlugin.java (hunk 2 above)
Comment (MrCredulous) [reply src/main/java/edu/cmu/cs/cs214/rec10/plugin/MoneyDoorPlugin.java]: Okay.

File: src/main/java/edu/cmu/cs/cs214/rec10/plugin/MoneyDoorPlugin.java (hunk 3 above)
Comment (MrCredulous) [reply src/main/java/edu/cmu/cs/cs214/rec10/plugin/MoneyDoorPlugin.java]: No problem.

File: src/main/java/edu/cmu/cs/cs214/rec10/plugin/MoneyDoorPlugin.java (hunk 1 above)
Comment (MrCredulous) [reply src/main/java/edu/cmu/cs/cs214/rec10/plugin/MoneyDoorPlugin.java]: Yeah you are right.

File: src/main/java/edu/cmu/cs/cs214/rec10/plugin/MoneyDoorPlugin.java [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+        }
+
+
+    }
```
Comment (lueliCMU) [inline src/main/java/edu/cmu/cs/cs214/rec10/plugin/MoneyDoorPlugin.java]: nothing

File: src/main/java/edu/cmu/cs/cs214/rec10/plugin/MoneyDoorPlugin.java (hunk 4 above)
Comment (MrCredulous) [reply src/main/java/edu/cmu/cs/cs214/rec10/plugin/MoneyDoorPlugin.java]: resolved.

**Pull request merged**